### PR TITLE
ENH: look at success state of status objects

### PR DESCRIPTION
**Pull request by tacaswell**
This is a proof-of-concept commit.

This relies on @klauer refactoring of the status objects.

### CODE REVIEW

**Comment by danielballan** [comment]
This effectively special-cases failure to move over other kinds of exceptions that can bubble up from ophyd. I'm not 100% convinced that it needs to be special-cased.

**Comment by tacaswell** [comment]
This catches async failures, the rest of them come up synchronously in the co-routines.   I think this is the only way to make sure that the exceptions BS needs come up on the correct threads / co-routines.

There are some epi-cycles with `self._task.cancel()` that I do not fully have pinned down.

Not sure what you mean by special casing failure to move, this also deals with trigger and kickoff failures (in the same way).

**Comment by danielballan** [comment]
I meant that is special cases a certain kind of failure coming up from the device layer. My impulse would be to just use exceptions. I see the `success` bit as something that debuggers will want and plans *might* use (to decide if it's important that something failed to move) but that the RunEngine doesn't need to know about.

**Comment by danielballan** [comment]
> I think this is the only way to make sure that the exceptions BS needs come up on the correct threads / co-routines.

If that's true though, my point is moot.

File: bluesky/run_engine.py [hunk 1]
```diff
@@ -1113,6 +1128,12 @@ def _wait(self, msg):
         if objs:
             yield from self._wait_for(Msg('wait_for', objs))
 
+    def _failed_status(self, ret):
+        '''
```
Comment (danielballan) [inline bluesky/run_engine.py]: Cool docstring, bro.

Comment (ericdill) [inline bluesky/run_engine.py]: Actually lol'd at this

**Pull request merged**